cli/parse_pdfs.py

```diff
@@ -146,25 +169,26 @@ def parse_file(
 
         all_pages_metadata.append(page_metadata)
 
-    document = ParserOutput(
-        document_id=input_task.document_id,
-        document_url=input_task.document_url,
-        document_name=input_task.document_name,
-        document_description=input_task.document_description,
-        document_content_type=input_task.document_content_type,
-        document_slug=input_task.document_slug,
-        pdf_data=PDFData(
-            page_metadata=all_pages_metadata,
-            md5sum=document_md5sum,
-            text_blocks=all_text_blocks,
-        ),
-    ).set_document_languages_from_text_blocks(min_language_proportion=0.4)
-
-    output_path = output_dir / f"{input_task.document_id}.json"
-
-    output_path.write_text(document.json(indent=4, ensure_ascii=False))
-
-    logging.info(f"Saved {output_path.name} to {output_dir}.")
+    if all_pages_metadata:
```

Comment:
We need to return an empty ParserOutput even if there's no detected text in the PDF, as we also need the parser to carry through metadata to the search indexer. This might need an `if/else` creating two different forms of the `ParserOutput`?

Reply:
@stefl14 from what i can see this is still an outstanding issue. let me know if i've missed something?

Reply:
You're right. Thought I solved it by removing the conditional having forgotten why I put it there, not realising that there would be no page metadata if the loop breaks early. I've moved the page dimensions and metadata code to the top of the loop and appended it to the metadata list before any early breaks.